src/charm.py

```diff
@@ -701,33 +701,63 @@ def _on_add_admin_user_action(self, event: ActionEvent) -> None:
         Args:
             event: Event triggering the add_admin_user action.
         """
-        email = event.params["email"]
-        password = event.params["password"]
         container = self.unit.get_container(CONTAINER_NAME)
-        if container.can_connect():
-            process = container.exec(
-                [
-                    os.path.join(DISCOURSE_PATH, "bin/bundle"),
-                    "exec",
-                    "rake",
-                    "admin:create",
-                ],
-                stdin=f"{email}\n{password}\n{password}\nY\n",
-                working_dir=DISCOURSE_PATH,
-                user=CONTAINER_APP_USERNAME,
-                environment=self._create_discourse_environment_settings(),
-                timeout=60,
+
+        email = event.params["email"]
+
+        if not container.can_connect():
+            event.fail("Unable to connect to container, container is not ready")
+            return
+        self._create_user(event)
+
+        process = container.exec(
+            [
+                os.path.join(DISCOURSE_PATH, "bin/bundle"),
+                "exec",
+                "rake",
+                "admin:promote",
+                email,
+            ],
+            working_dir=DISCOURSE_PATH,
+            user=CONTAINER_APP_USERNAME,
+            environment=self._create_discourse_environment_settings(),
+        )
+        try:
+            process.wait_output()
+            event.set_results(email)
+        except ExecError as ex:
+            event.fail(
+                f"Failed to make user with email {email} an admin: {ex.stdout}"  # type: ignore
             )
-            try:
-                process.wait_output()
-                event.set_results({"user": f"{email}"})
-            except ExecError as ex:
-                event.fail(
-                    # Parameter validation errors are printed to stdout
-                    f"Failed to create user with email {email}: {ex.stdout}"  # type: ignore
-                )
-        else:
-            event.fail("Container is not ready")
+
+    def _create_user(self, event: ActionEvent):
+        """Create a new user in Discourse.
```

Comment:
Maybe this function could receive an email and a password and not an event, and so be more generic. The only use of the event is to set the action to failed, which maybe it could be in the calling function.

Reply:
@codethulu can you post a short comment before resolving a conversation? that applies for all the resolved convos. Thanks!